### ISSUE TITLE
The new Reactive.jl

### ISSUE THREAD

**Issue by timholy** [jverzani/GtkInteract.jl#5]
https://github.com/JuliaLang/Reactive.jl/pull/65 was a big (and likely breaking) change that just got merged. Once it gets tagged (CC @shashi), this will likely need updating. I might be able to help, but first some questions:

- Are you interested in registering this sometime? If so, I might use it in [GtkUtilities.jl](https://github.com/timholy/GtkUtilities.jl).
- Given the existence of [Immerse.jl](https://github.com/JuliaGraphics/Immerse.jl), what do you think of removing the Winston dependency? I went with Gadfly mostly because it is more actively maintained.

**Comment by jverzani** [comment]
Thanks for the heads up Tim. I'll see if I can do the updates. Hopefully it is straightforward. Once done, let's talk about if it makes sense to merge into GtkUtilities. I'd be all for that if it made sense to you. As for Winston, absolutely. I had it only because I couldn't get Gadfly to render quickly enough -- but you did that hard work. I'll ping you when I get this done, hopefully this week.

**Comment by timholy** [comment]
To be clear, I wasn't suggesting merging it in; the symmetry with regards to `Interact.jl` suggests leaving this a standalone package. I was more thinking about `using` it in GtkUtilities.

Depending on my schedule, I may be able to help with the work, though obviously great if you can do it.

If you're not already aware of it, one thing I'd like to call your attention to: https://github.com/JuliaLang/Gtk.jl/issues/161 (similarly https://github.com/JuliaLang/Gtk.jl/issues/167), and see workarounds in places [like this](https://github.com/timholy/GtkUtilities.jl/blob/b36edfca6b4f0c0b6351f1c1409e4e2d04ca4f8f/src/link.jl#L243-L253). See also the new `@guarded` macro for callbacks to keep from tanking Gtk when a callback function has an error. Most comprehensive docs on all this are [here](https://github.com/JuliaLang/Gtk.jl/blob/master/doc/more_signals.md).

**Comment by timholy** [comment]
Gadfly is still a bit slow for the first plot, but the precompile directives help a lot. I'm also contemplating providing support for the `userimg.jl` solution, which should make the latency-to-first-plot nearly zero.

**Comment by shashi** [comment]
You should be good to go if you follow the deprecation warnings given by Reactive master. Although this one: `WARNING: Reactive.Input is deprecated, use Reactive.Node instead` should have been `WARNING: Reactive.Input is deprecated, use Reactive.Signal instead`. `Signal` is exported and is a typealias to `Node`, which is not exported. ;)

**Comment by jverzani** [comment]
Thanks Tim, I'll have  a look a those links to the Gtk issues. I haven't followed the development there for a while. I'll see how far I get before bothering you with questions. I know you are super busy.

**Comment by jverzani** [comment]
Hi @shahsi. Quick question. In the master version of `Interact`, I see: `abstract Widget <: SignalSource` which gives an error as there is no SignalSource abstract type to be found. I think this was in `Reactive`, but has been renamed. Is there a newer version of `Interact` I should be using, besides master from https://github.com/JuliaLang/Interact.jl ?

**Comment by shashi** [comment]
Yeah SignalSource is removed.

I have made these changes to interact but unable to test it (due to an
issue with my IJulia). Will try harder to test and ping you once I have a
working version. Sorry I didn't think of this before encouraging you to
work on this issue !

**Comment by jverzani** [comment]
Could your changes be pushed to a different branch? Then I could test
things out which you patch up your ijulia.

On Tue, Nov 17, 2015 at 8:07 AM, Shashi Gowda <notifications@github.com>
wrote:

> Yeah SignalSource is removed.
>
> I have made these changes to interact but unable to test it (due to an
> issue with my IJulia). Will try harder to test and ping you once I have a
> working version. Sorry I didn't think of this before encouraging you to
> work on this issue !
>
> On Tue, Nov 17, 2015, 6:19 PM john verzani <notifications@github.com>
> wrote:
>
> > Hi @shahsi. Quick question. In the master version of Interact, I see:
> abstract
> > Widget <: SignalSource which gives an error as there is no SignalSource
> > abstract type to be found. I think this was in Reactive, but has been
> > renamed. Is there a newer version of Interact I should be using, besides
> > master from https://github.com/JuliaLang/Interact.jl ?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/jverzani/GtkInteract.jl/issues/5#issuecomment-157362877
> >
> > .
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jverzani/GtkInteract.jl/issues/5#issuecomment-157365787>
> .
>



-- 
John Verzani
Chair, Department of Mathematics
College of Staten Island, CUNY
verzani@math.csi.cuny.edu

**Comment by timholy** [comment]
Really nice work!

I'm going to start playing with this as a foundation for the new ImageView. I am probably going to try to avoid needing any plotting package in ImageView, so the dependency on Plots could be a little weird, but AFAICT that's decoupled from actually loading a plotting package. So I doubt it will cause trouble.

**Comment by jverzani** [comment]
I realize the Plots dependency it totally optional, so it is now removed. I'm waiting on a clarification on how the new Reactive and Interact play together, and once that is cleared up will add this to METADATA. Thanks for the encouragement.
